api/src/routing.js

```diff
@@ -158,7 +159,8 @@ if (process.argv.slice(2).includes('--allow-cors')) {
 
 app.use((req, res, next) => {
   req.id = uuid.v4();
-  next();
+  req.headers[serverUtils.REQUEST_ID_HEADER] = req.id;
+  asyncLocalStorage.run({ clientRequest: req }, () => next());
```

Comment:
Feels bloated to store the whole `req` here. Why not just the req.id?

Reply:
I'd have to trace it through - it'd be nice if req could be GCed earlier but this may not be possible...

Reply:
I don't think it can get GC-ed before sending a response to the client, which is when the async local storage also "expires"